src/dotnet/APIView/APIViewWeb/Client/src/shared/signalr.ts

```diff
@@ -58,8 +53,270 @@ $(() => {
     hp.updateCommentThread(commentsRow, partialViewResult);
     hp.addCommentThreadNavigation();
     hp.removeCommentIconIfEmptyCommentBox(elementId);
+
+    let size = 28;
+    let $navLinks = $("nav.navbar a.nav-link");
+    let username;
+
+    for (let nav of $navLinks) {
+      if (nav.innerText.includes("Profile")) {
+        let href = nav.getAttribute("href");
+        if (href) {
+          let hrefString: string = href;
+          let hrefSplit = hrefString.split("/");
+          username = hrefSplit[hrefSplit.length - 1];
+        }
+      }
+    }
+
+    let url: string = "https://github.com/" + username + ".png?size=" + size;
+    $("div.review-thread-reply div.reply-cell img.comment-icon").attr("src", url);
   });
 
+  let approvalPendingText = "Current Revision Approval Pending";
+  let approvedByText = "Approved by:";
+  let approvesCurrentRevisionText = "Approves the current revision of the API";
+
+  connection.on("ReceiveApprovalSelf", (reviewId, revisionId, approvalToggle) => {
+    hp.disableButtonTemp("form.form-inline button.btn", 5000);
```

Comment:
Try to disable only the specific instance of the button.

Reply:
Not sure if this is possible - comment update uses event listeners so we can use `e.target`, but approvals use cshtml.cs. is there a similar version for `<form>` in Razor?

Reply:
Or should I add a new class to the form/button on razor then make an event listener on review.ts?